ShellPkg/Library/UefiShellDebug1CommandsLib/EfiCompress.c

```diff
@@ -79,10 +79,15 @@ ShellCommandRunEfiCompress (
       ShellStatus = SHELL_INVALID_PARAMETER;
     } else {
       TempParam = ShellCommandLineGetRawValue (Package, 1);
-      ASSERT (TempParam != NULL);
+      if (TempParam == NULL) {
+        ASSERT (TempParam != NULL);
+        ShellPrintHiiEx (-1, -1, NULL, STRING_TOKEN (STR_GEN_PARAM_INV), gShellDebug1HiiHandle, L"eficompress");
+        ShellStatus = SHELL_INVALID_PARAMETER;

```

Comment:
Should we avoid to continue executing the function and have a goto to the end of the function ?
Here the following would still be exectured:
```
 InFileName  = ShellFindFilePath (TempParam);
 ...
```

Reply:
Added goto and also logic to call ShellCommandLineFreeVarList if required.